stingray/tests/test_io.py

```diff
@@ -163,132 +163,6 @@ def test_operation(self):
 
 class TestFileFormats(object):
 
-    def test_pickle_read_write(self):
-        test_object = TmpIOReadWrite()
-        write(test_object, filename='test.pickle', format_='pickle')
-        assert read('test.pickle', 'pickle') is not None
-        os.remove('test.pickle')
-
-    def test_pickle_attributes(self):
-        """Test if pickle maintains class object attributes."""
-        test_object = TmpIOReadWrite()
-        write(test_object, filename='test.pickle', format_='pickle')
-        rec_object = read('test.pickle', 'pickle')
-        assert rec_object.number == test_object.number
-        assert rec_object.str == test_object.str
-        assert rec_object.list == test_object.list
-        assert (rec_object.array == test_object.array).all()
-        assert rec_object.long_number == test_object.long_number
-        assert (rec_object.long_array == test_object.long_array).all()
-
-        os.remove('test.pickle')
-
-    def test_pickle_functions(self):
-        """Test if pickle maintains class methods."""
-        test_object = TmpIOReadWrite()
-        write(test_object,'test.pickle', 'pickle')
-        assert read('test.pickle', 'pickle').test_operation() == test_object.number * 10
-        os.remove('test.pickle')
-
-    @skip_condition
-    def test_hdf5_write(self):
-        test_object = TmpIOReadWrite()
-        write(test_object, 'test.hdf5', 'hdf5')
-        os.remove('test.hdf5')
-
-    @skip_condition
-    def test_hdf5_read(self):
-        test_object = TmpIOReadWrite()
-        write(test_object, 'test.hdf5', 'hdf5')
-        read('test.hdf5','hdf5')
-        os.remove('test.hdf5')
-
-    @skip_condition
-    def test_hdf5_data_recovery(self):
-        test_object = TmpIOReadWrite()
-        write(test_object, 'test.hdf5', 'hdf5')
-        rec_object = read('test.hdf5','hdf5')
-
-        assert rec_object['number'] == test_object.number
-        assert rec_object['str'] == test_object.str
-        assert (rec_object['list'] == test_object.list).all()
-        assert (rec_object['array'] == np.array(test_object.array)).all()
-        assert rec_object['long_number'] == np.double(test_object.long_number)
-        assert (rec_object['long_array'] == np.double(np.array(test_object.long_array))).all()
-
-        os.remove('test.hdf5')
-
-    def test_save_ascii(self):
-        time = [1, 2, 3, 4]
-        counts = [2, 3, 41, 4]
-
-        write(np.array([time, counts]).T, "ascii_test.txt",
-              "ascii")
-
-        os.remove("ascii_test.txt")
-
-    def test_save_ascii_with_mixed_types(self):
-        time = ["bla", 1, 2, 3]
-        counts = [2,3,41,4]
-        with pytest.raises(Exception):
-             write(np.array([time, counts]).T,
-                   "ascii_test.txt", "ascii")
```

Comment:
I feel like not all of these tests made it into the new test class for `base.py`, but some  might be important, like this one above. I would suggest going through this list of tests and making sure those that test edge cases make it into the new test classes for `base.py`.

Reply:
All of these corner cases, plus many more, are included in the Astropy I/O interface. Also, the presence of bad data is tested internally in all these classes, besides the read/write interface